src/client/datascience/jupyterExecution.ts

```
@@ -59,8 +59,15 @@ class JupyterKernelSpec implements IJupyterKernelSpec {
         this.specFile = file;
     }
     public dispose = () => {
-        if (this.specFile && IsGuidRegEx.test(path.basename(path.dirname(this.specFile)))) {
-            fs.removeSync(path.dirname(this.specFile));
+        if (this.specFile &&
+            IsGuidRegEx.test(path.basename(path.dirname(this.specFile)))) {
+            try {
+                fs.removeSync(path.dirname(this.specFile));

```

Please make this async.

---

It can't be. The dispose can't return anything.

---
In reply to: [240364388](https://github.com/Microsoft/vscode-python/pull/3614#discussion_r240364388) [](ancestors = 240364388)

---

Although I could just ignore errors on it.

---
In reply to: [240369636](https://github.com/Microsoft/vscode-python/pull/3614#discussion_r240369636) [](ancestors = 240369636,240364388)

---

> It can't be. The dispose can't return anything.

A number of async `dispose` methods already exist in DS code, how is this any different?

---

We must always avoid sync operations. Specially here, perfoming I/O operations in dispose slows down overall dispose method time.